### PR TITLE
Add _thread.start_new_thread

### PR DESCRIPTION
**Pull request by coolreader18**
~~Dependent on #1933 and #1935~~

### CODE REVIEW

**Comment by coolreader18** [comment]
`test_threading` does some of that already, but I think I could probably verify the ordering now/make sure that it's actually concurrent somehow.

**Comment by coolreader18** [comment]
@palaviv I was actually able to get `test.test_thread` passing with not too much effort, so I think that should work for tests. I guess the `_thread` module really doesn't have that much API surface.

File: vm/Cargo.toml [hunk 1]
```diff
@@ -71,7 +71,7 @@ smallbox = "0.8"
 bstr = "0.2.12"
 crossbeam-utils = "0.7"
 generational-arena = "0.2"
-parking_lot = { git = "https://github.com/Amanieu/parking_lot" }
+parking_lot = { git = "https://github.com/coolreader18/parking_lot", branch="raw-re-is_locked" }
```
Comment (palaviv) [inline vm/Cargo.toml:74]: I was a bit iffy when we added the `parking_lot` from Github. For now at least add a TODO to change this.

Comment (coolreader18) [reply vm/Cargo.toml:74]: That was just temporary, the PR for that is merged now. I'll still add a TODO for the official git version, though.

**Pull request merged**